packages/grafana-ui/src/components/Link/TextLink.mdx

```diff
@@ -1,10 +1,152 @@
-import { Meta, Preview, ArgTypes } from '@storybook/blocks';
+import { Meta, Props, Preview, ArgsTable } from '@storybook/addon-docs/blocks';
 import { TextLink } from './TextLink';
+import { Basic } from './TextLink.story.tsx';
 
 <Meta title="MDX|TextLink" component={TextLink} />
 
 # TextLink
 
 The TextLink component renders an anchor tag `<a>` that takes users to another page, external or internal to Grafana.
 
-<ArgTypes of={TextLink} />
+---
+
+In this documentation you can find:
+
+1. [Usage](#usage)
+1. [Types](#types)
+1. [Behaviour](#behaviour)
+   <br />
+   4.1. [Inline](#inline-behaviour)
+   <br />
+   4.2. [Standalone](#standalone-behaviour)
+   <br />
+   4.3. [Icons](#icons-behaviour)
+1. [Props table](#propstable)
+1. [Related](#related)
+
+<br />
+<br />
+
+## <a name="usage"/> Usage
+
+<br />
+
+### **When to use**
+
+- To redirect the user to another page, external or internal
+- When redirecting the user through a link in a word or phrase within a paragraph.
+- When redirecting the user through a link in a word or phrase, not within a paragraph but surrounded by other blocks of text.
+
+### **When not to use**
+
+- When its purpose is to trigger an action, please use a Button or IconButton component
+- When it still redirects the user but it is not surrounded by text - for example, being in a row with more buttons -, please use the LinkButton component.
+- When the link is just comprised by an icon use IconButton component.
+
+### **Do’s**
+
+- Use the default styles as the first and main option as they are though to be accessible and consistent.
+- Use the external icon (the default one) when the TextLink goes to an external page as its meaning is broadly known.
+- In case the default icon does not fit, please use one of the icons of our [library](https://developers.grafana.com/ui/latest/index.html?path=/story/docs-overview-icon--icons-overview).
+
+### **Don’ts**
+
+- Do not use ‘primary’ color in a standalone link unless it is a special use case such as the ‘Dashboard list’, menu items or similar.
+- Do not override any styles in any way. Instead, ask the Design System team for an improvement of this component.
+- Do not attempt to add a new icon. You should fulfil a contribution path to first add the icon to our Design System and then use it in this component.
+
+  <br />
+  <br />
+
+## <a name="types"/> Types
+
+Within Grafana, there are two different types of _text links_, 'inline' or 'standalone', whether they are wrapped by text or being a standalone element.
+
+- **Inline**: It is part of a broader text. It will be surrounded by text, before, after or in both directions. Although it can show an icon at its end, we encouragedly advice not to do that.
```

Comment:
```suggestion
- **Inline**: It is part of a broader text. It will be surrounded by text, before, after or in both directions. Although it can show an icon at its end, we advise not to do that.
```

Not sure 'encouragedly' is a word 😅